test/stream.test.js

```diff
@@ -136,13 +136,29 @@ describe('StreamAPI', () => {
           await client.query(q.Update(doc.ref, {}))
         })
         .on('error', error => {
+          checkErrorCallback(error)
+          done()
+        })
+        .start()
+    }
+
+    test('wraps error events', async done => {
+      testErrorEvent(done, error => {
+        if (error.code && error.description) {
           expect(error.code).toEqual('permission denied')
           expect(error.description).toEqual(
             'Authorization lost during stream evaluation.'
           )
-          done()
-        })
-        .start()
+        }
+      })
+    })
+
+    test('wraps delegated error events if stream closed unexpectedly', async done => {
+      testErrorEvent(done, error => {
+        if (error.code === undefined) {
```

Comment:
what happens if `error.code !== undefined`? what should be the assertion in this case?

Reply:
i wanted to test the error event that happens when we destroy the secret while the stream connection is already opened:
https://github.com/parkhomenko/faunadb-js/blob/drv-444/test/stream.test.js#L129-L142

the issue was that in that case, the ".error" event triggered 2 times:
- the error object is sent like this: {code: 'permission denied', description: 'Authorization lost during stream evaluation.'}
- the internal `onEnd` event is triggered and delegates to the 'onError' event: https://github.com/parkhomenko/faunadb-js/blob/drv-444/src/stream.js#L187-L193, and we have a different `error` object, that doesn't have fields `code` and `description` in it

the first test checks the first case: https://github.com/parkhomenko/faunadb-js/blob/drv-444/test/stream.test.js#L145-L154 (`in this case error.code !== undefined`)
the second test checks that we got the `error` event delegated from `onEnd` event, and `error.code` is undefined in this scenario

so if `error !== undefined` the assertion defined here: https://github.com/parkhomenko/faunadb-js/blob/drv-444/test/stream.test.js#L148-L151 checks the `error` object

Reply:
I understand your arguments, but the conditional assertion on tests looks wrong to me. I would like to know others opinion regarding this.

Reply:
could you advise who from the team can I add to this PR to take a look at it?

Reply:
@erickpintor @evbruno @rodrimaia can you take a look on this?

Reply:
I'm okay with the error type. I think the assertion could be a little bit better.

```
if (error instanceof TypeError) {
  expect(error.message).toEqual('network error')
}
```

Reply:
Good point. I've changed it as you suggested. Thanks!